pkg/api/volume.go

```diff
@@ -74,6 +74,18 @@ func (this *VolumePortal) CreateVolume() {
 
 	this.Ctx.Output.SetStatus(StatusAccepted)
 	this.Ctx.Output.Body(body)
+
+	// Call global controller variable to handle create volume request.
+	var errchan = make(chan error, 1)
+	go controller.Brain.CreateVolume(c.GetContext(this.Ctx), result, pol, profile, errchan)
+	if err := <-errchan; err != nil {
+		reason := fmt.Sprintf("Marshal volume created result failed: %s", err.Error())
+		this.Ctx.Output.SetStatus(model.ErrorBadRequest)
+		this.Ctx.Output.Body(model.ErrorBadRequestStatus(reason))
+		log.Error(reason)
+		return
+	}
+
 	return
```

Comment:
So this CreateVolume function doesn't return after AsynCreateVolume call?  How is this really async?

Does controller.Brain.CreateVolume wait until a volume is really created?

This comment applies to all other "async" functions.